### PR TITLE
Feature/sof-6588 feat: add checks for overlapping atoms etc, highlight in basis viewer and in wave viewer

### PR DESCRIPTION
**Pull request by VsevolodX**

### CODE REVIEW

File: src/material.js [hunk 1]
```diff
@@ -49,4 +49,68 @@ export class Material extends Made.Material {
...
         return new Material(config);
     }
+
+    // eslint-disable-next-line class-methods-use-this
+    getBasisChecks() {
+        const { basis } = this.toJSON();
+        const checks = [];
+
+        // TODO: implement testing for rules in Material class (warnings mixin?)
+        for (let i = 0; i < basis.elements.length; i++) {
```
Comment (zoomchik) [inline src/material.js]: May be we can simplify it a bit to eliminate nested loops, something like this:
`basis.elements.map((el) => JSON.stringify(el)).forEach((e, i, a) => {
         const index = a.indexOf(e);
         if (index !== i) checks.push(...);
     });`

File: tests/cucumber/features/3d-editor/multiple-selection-cancel.feature [hunk 2]
```diff
@@ -4,7 +4,7 @@ Feature: User can cancel active multiple selection and all changes should be rev
     When I open materials designer page
     And I create materials with the following data
       | name      | basis             |
-      | supercell | Si 0 0 0;Si 0 0 0 |
+      | supercell | Si 0 0 0;Si 0 0 1 |
```
Comment (timurbazhirov) [inline tests/cucumber/features/3d-editor/multiple-selection-cancel.feature]: 0.25 0.25 0.25 here and below

File: src/components/source_editor/BasisText.jsx [hunk 3]
```diff
@@ -24,10 +26,13 @@ class BasisText extends React.Component {
...
     // eslint-disable-next-line no-unused-vars
     UNSAFE_componentWillReceiveProps(nextProps, nextContext) {
-        const { content } = this.props;
+        const { content, checks } = this.props;
         if (content !== nextProps.content) {
             this.reformatContentAndUpdateStateIfNoManualEdit(nextProps.content);
         }
+        if (checks !== nextProps.checks) {
+            this.setState({ checks: nextProps.checks });
+        }
```
Comment (timurbazhirov) [inline src/components/source_editor/BasisText.jsx:34]: Same as https://github.com/Exabyte-io/cove.js/pull/41/files#r1378311003

Comment (VsevolodX) [reply src/components/source_editor/BasisText.jsx:34]: Got it

**Comment by VsevolodX** [comment]
Still waiting for made.js update

File: entrypoint.sh [hunk 4]
```diff
@@ -1,8 +1,5 @@
 #!/usr/bin/env bash
 
-source $NVM_DIR/nvm.sh
-nvm use ${NODE_VERSION}
-
```
Comment (timurbazhirov) [inline entrypoint.sh]: Remove

**Review by chu-k — APPROVED** [review]
Verified functionality locally. Commenting as reminder to add cucumber test for checking 'warning' tag, tbd in a follow on task perhaps.

**Pull request merged**